### DISCUSSION TITLE
A CodeEditor Widget

### DISCUSSION THREAD

**Discussion by SKbarbon** [flet-dev/flet#4903] [Extensions] [answered]
This is a CodeEditor widget in `flutter`, I really like it and I think it will add a great widget to `flet` if `flet` did support it.

Its a multi-line TextEditor, but prepared to be a programming code editor.

```
Flutter Code Editor is a multi-platform code editor supporting:

- Syntax highlighting,
- Code blocks folding,
- Autocompletion,
- Read-only code blocks,
- Hiding specific code blocks,
- Themes,
- And many other features.
```

![Basic example](https://raw.githubusercontent.com/akvelon/flutter-code-editor/main/example/images/main.gif)

You can take a look at it here:

https://github.com/akvelon/flutter-code-editor

**Comment by Bronya0** [comment]
+1

**Comment by qwerty108109** [comment]
This would be so awesome. I wonder if I would have to build a package in Flutter and have it imported because I do not think Flutter has a native code editing widget.  But this is something I really want.

**Comment by ndonkoHenri** [comment, marked as answer]
Implemented in #6162, and will be available as from v0.80.6
Docs: https://docs.flet.dev/codeeditor/
